src/components/Editor/Overlay.tsx

```diff
@@ -94,60 +117,90 @@ function Overlay(): JSX.Element {
   } = useActionList(experienceData);
 
   // useEffect for autoSave
-  useEffect(()=>{
+  useEffect(() => {
     experienceData.experience.actionList = [...actionList];
     saveExp(userId, experienceData.experience);
-  }, [actionList, experienceData, userId])
+  }, [actionList, experienceData, userId]);
+
+  const renderTool = () => {
+    if (toolDoc)
+      return (
+        <ToolDocGrid>
+          <ToolDocItem />
+        </ToolDocGrid>
+      );
+    return <div />;
+  };
+
+  const renderText = () => {
+    if (textBox)
+      return (
+        <TextEditorGrid>
+          {" "}
+          <TextEditor />
+        </TextEditorGrid>
+      );
+    return <div />;
+  };
+
```

Comment:
are they possible or necessary to be created from another file there for it will be easier the do the render test if needed and cleaner easier for this code?